src/Connectors/src/Connector.EFCore/MySqlDbContextOptionsExtensions.cs

```diff
@@ -187,10 +188,20 @@ private static DbContextOptionsBuilder DoUseMySql(DbContextOptionsBuilder builde
         }
         else
         {
-            // If the server version wasn't passed in, use the EF Core lib to autodetect it (this is the part that creates an extra connection)
-            serverVersion ??= ReflectionHelpers.FindMethod(EntityFrameworkCoreTypeLocator.MySqlVersionType, "AutoDetect", new[] { typeof(string) }).Invoke(null, new object[] { connection });
+            // In 6.0, Pomelo requires server version but MySql.Data does not support it as a param. Look for a method that requires it first
             useMethod = FindUseSqlMethod(extensionType, new[] { typeof(DbContextOptionsBuilder), typeof(string), EntityFrameworkCoreTypeLocator.MySqlVersionType, typeof(Action<DbContextOptionsBuilder>) });
-            parameters = new[] { builder, connection, serverVersion, mySqlOptionsAction };
+
+            if (useMethod is null)
+            {
+                useMethod = FindUseSqlMethod(extensionType, new[] { typeof(DbContextOptionsBuilder), typeof(string), typeof(Action<DbContextOptionsBuilder>) });
+                parameters = new[] { builder, connection, mySqlOptionsAction };
+            }
+            else
+            {
+                // If the server version wasn't passed in, see if we need to use the EF Core lib to autodetect it (this is the part that creates an extra connection)
+                serverVersion ??= ReflectionHelpers.FindMethod(EntityFrameworkCoreTypeLocator.MySqlVersionType, "AutoDetect", new[] { typeof(string) }).Invoke(null, new object[] { connection });
+                parameters = new[] { builder, connection, serverVersion, mySqlOptionsAction };
+            }
```

Comment:
I'm not confident asserting the correctness of this, because I don't sufficiently know what's going on here. I trust you did the right thing, but if you're in need of an extra pair of eyes, please ask someone on the team that has a better understanding of this.

Reply:
I got around to testing this with a sample app.. the code is cleaner now and works with 6.0 versions of both packages we're intending to support

Reply:
That's great! Closing this one.